src/VirtoCommerce.CustomerExportImportModule.Data/Validation/ImportAddressValidator.cs

```diff
@@ -29,21 +28,28 @@ private void AttachValidators()
                 , () =>
                 {
                     RuleFor(x => x.Record.AddressType).IsEnumName(typeof(AddressType))
+                        .NotEmpty()
+                        .WithMissingRequiredValueCodeAndMessage("Address Type")
                         .When(x => !string.IsNullOrEmpty(x.Record.AddressType))
                         .WithInvalidValueCodeAndMessage("Address Type")
                         .WithImportState();
 
                     RuleFor(x => x.Record.AddressFirstName)
+                        .NotEmpty()
+                        .WithMissingRequiredValueCodeAndMessage("Address First Name")
                         .MaximumLength(128)
                         .WithExceededMaxLengthCodeAndMessage("Address First Name", 128)
                         .WithImportState();
                     RuleFor(x => x.Record.AddressLastName)
+                        .NotEmpty()
+                        .WithMissingRequiredValueCodeAndMessage("Address Last Name")
                         .MaximumLength(128)
                         .WithExceededMaxLengthCodeAndMessage("Address Last Name", 128)
                         .WithImportState();
 
                     RuleFor(x => x.Record.AddressEmail)
                         .EmailAddress()
+                        .When(x => !string.IsNullOrEmpty(x.Record.AddressEmail))
```

Comment:
```js
.When(x => !string.IsNullOrEmpty(x.Record.AddressEmail))
.EmailAddress()
```
otherwise check will run, but results will not be taken into account if email is null or empty

Reply:
It is not that. https://docs.fluentvalidation.net/en/latest/conditions.html#